[DSS-297] Hero: Design Audit + New Variant

pixelflips opened the pull request:
## Description
Updates hero component to include new variant and related UI changes based on design audit.

- Adds new contained variant to support non-full-bleed images.
- Updates the color of the play icon (charcoal).
- Removed branding prop from React version. (Could find no usage)
- Various minor adjustments and fixes to sync with the latest design spec.
- Documentation updates for a more wide array of examples.
- Adds a link to the React component from docs.


## Screenshots
<!-- OPTIONAL(recommended): Show any visual updates -->
|  Before  |  After  |
|--------|--------|
|<img width="1103" alt="Screenshot 2023-03-14 at 11 09 15 AM" src="https://user-images.githubusercontent.com/1175111/225100518-52d81d32-a8d3-47e7-9bcb-2e37265008a8.png">|<img width="1093" alt="Screenshot 2023-03-14 at 11 08 47 AM" src="https://user-images.githubusercontent.com/1175111/225100585-edf1d81a-d6a4-499b-8afe-033187ce4997.png">|


## Testing in `sage-lib`

**Rails/Docs**

- Navigate to [Hero](http://localhost:4000/pages/component/hero?tab=preview) 
- Verify the updated design tweaks & the new contained variant

**React/Storybook**

- Navigate to [Hero](http://localhost:4100/?path=/story/sage-hero--default) 
- Verify the updated design tweaks & the new contained variant


## Testing in `kajabi-products`

1. (**MEDIUM**) Updates hero to match latest design spec and adds contained variant.
   - [ ] Dashboard
   - [ ] Creator Tools > Projects List


## Related
DSS-297

QuintonJason commented on packages/sage-react/lib/Hero/Hero.jsx:
```diff
@@ -2,13 +2,14 @@ import React from 'react';
 import classnames from 'classnames';
 import { Button } from '../Button';
-import { HERO_SIZES } from './configs';
+import { HERO_SIZES, } from './configs';
```
Can you remove this `,`?

pixelflips replied:
Done.

QuintonJason commented on packages/sage-assets/lib/stylesheets/components/_hero.scss:
```diff
@@ -58,30 +50,34 @@ $-hero-play-icon-background-color-hover: rgba(sage-color(white, 100), 1);
 }
 
+.sage-hero--large.sage-hero--contained {
+
```
Remove this line break

QuintonJason commented on packages/sage-assets/lib/stylesheets/components/_hero.scss:
```diff
@@ -122,20 +123,22 @@ $-hero-play-icon-background-color-hover: rgba(sage-color(white, 100), 1);
+  .sage-hero--contained & {
+    object-fit: unset;
+    height: auto;
+    max-height: rem(235px);
```
Can you put this value in a var?

QuintonJason commented on packages/sage-assets/lib/stylesheets/components/_hero.scss:
```diff
@@ -91,25 +87,30 @@ $-hero-play-icon-background-color-hover: rgba(sage-color(white, 100), 1);
-    border-bottom-right-radius: 0;
+  .sage-hero--contained & {
+    height: auto;
+    margin: rem(36px);
```
Can this one use the margin var from above?

QuintonJason reviewed (changes requested):
This changes look great! I'm requesting the React side also be more aligned with respect to demos. On the Rails side we have a bunch:
- Default
- Contained
- w/ CTA
- Contained w/ CTA
- Borderless with Custom Background

Would you be able to add the following stories?
- Default
- Contained
- w/ CTA
- Borderless with Custom Background

QuintonJason reviewed (approved):
Looks and works great!

anechol commented on packages/sage-assets/lib/stylesheets/components/_hero.scss:
```diff
@@ -56,32 +45,27 @@ $-hero-play-icon-background-color-hover: rgba(sage-color(white, 100), 1);
 }
 
 .sage-hero__title {
   @extend %t-sage-heading-4;
```
Should this be heading-3?

anechol commented on packages/sage-assets/lib/stylesheets/components/_hero.scss:
```diff
@@ -56,32 +45,27 @@ $-hero-play-icon-background-color-hover: rgba(sage-color(white, 100), 1);
   }
 }
 
 .sage-hero__body {
```
Can we update this to match the design spec? The font-weight says 450 in the design file for this text, but it should be `medium` as medium = book. It's a long story...

anechol commented on packages/sage-assets/lib/stylesheets/components/_hero.scss:
```diff
@@ -56,32 +45,27 @@ $-hero-play-icon-background-color-hover: rgba(sage-color(white, 100), 1);
 .sage-hero__body {
   grid-area: content;
+  padding: 0 sage-spacing(md);
   margin-bottom: $-hero-margin-y;
```
Should this be 24px?

anechol replied:
I'm still seeing 32px here

pixelflips replied:
@anechol should be good to go now. Thanks so much for catching those!

anechol commented on packages/sage-react/lib/Hero/Hero.jsx:
```diff
@@ -34,35 +40,27 @@ export const Hero = ({
+        )}
+        {footerActions && (
+          <div className="sage-hero__footer-actions">
+            <Button.Group gap="sm">
```
Should this be md?

anechol reviewed (changes requested):
Nice work here! Just added some comments for design spec stuff.

pixelflips commented:
> Nice work here! Just added some comments for design spec stuff.

@anechol thanks so much for catching those! I was so focused on the new image/layout I missed some of those smaller details. It's a big yes to all of those and I've updated them. Thanks again!

anechol reviewed (changes requested):
Thanks for making those changes. Just a couple more small things. It looks like the button gap between the CTAs should be 24px over 16px

anechol reviewed (approved):
LGTM!! 🚀
